tests/integration_tests/test_collection_embedding_function.py

```
@@ -384,6 +390,454 @@ def test_get_or_create_collection_both_none_error(self, db_client):
+    def test_unregistered_embedding_function_error(self, db_client):
+        """
+        Test that getting a collection with unregistered embedding function raises error.
+
+        Tests:
+        - Create collection with unregistered embedding function (should fail or not persist)
+        - Attempt to get collection should handle gracefully
+
+        Automatically runs for: embedded, server, oceanbase
+        """
+        collection_name = f"test_ef_unregistered_{int(time.time() * 1000)}"
+        print(f"\n✅ Testing unregistered embedding function handling")
```

Ruff (0.14.13)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)